api/valor_api/backend/metrics/text_generation.py

```diff
@@ -0,0 +1,529 @@
+        or len(weights) == 0
+    ):
+        raise ValueError(
+            "Received incorrect inputs. predictions should be a string, references a list of strings, and weights a list/tuple of floats"
+        )
+
+    # handle case where user passes in a single prediction
+    if isinstance(predictions, str):
+        processed_predictions = [predictions]
+        processed_references = [references]
+        # handle case where user passes multiple predictions
+    elif isinstance(predictions, list) and all(
+        [isinstance(lst, list) for lst in references]
+    ):
+        processed_predictions = predictions
+        processed_references = references
+    else:
+        raise ValueError(
+            "prediction should be a str or list[str]. If prediction is a list[str], then references must be a list of lists."
+        )
+
+    output = defaultdict(float)
+    tokenizer = RegexpTokenizer(
+        r"\w+|\$[\d]+|[^\s\.]+"
+    )  # regex tokenizer that ignores periods
+
+    for pred, refs in zip(processed_predictions, processed_references):
+
+        tokenized_prediction = tokenizer.tokenize(pred)
+        tokenized_references = [tokenizer.tokenize(ref) for ref in refs]
+
+        # find the max value for each prediction
+        output[pred] = max(
+            bleu_score.sentence_bleu(
+                references=tokenized_references,
+                hypothesis=tokenized_prediction,
+                weights=weights,
+            ),  # type: ignore
+            output[pred],
+        )
+
+    return [
+        {"prediction": key, "value": value} for key, value in output.items()
+    ]
+
+
+def _setup_llm_client(
+    llm_api_params: dict[str, str | dict],
+) -> LLMClient:
+    """
+    Setup an LLM client for LLM guided evaluation.
+
+    Parameters
+    ----------
+    llm_api_params : dict[str, str | dict], optional
+        The parameters to setup the client with.
+
+    Returns
+    ----------
+    LLMClient
+        A wrapper for other LLM API clients.
+    """
+    assert (
+        "client" in llm_api_params or "api_url" in llm_api_params
+    ), "Need to specify the client or api_url."
+    assert not (
+        "client" in llm_api_params and "api_url" in llm_api_params
+    ), "Cannot specify both client and api_url."
+
+    if llm_api_params.get("client") is not None:
+        if llm_api_params["client"] == "openai":
+            client_cls = WrappedOpenAIClient
+        elif llm_api_params["client"] == "mistral":
+            client_cls = WrappedMistralAIClient
+        elif llm_api_params["client"] == "mock":
+            client_cls = MockLLMClient
+        else:
+            raise ValueError(
+                f"Client {llm_api_params['client']} is not supported."
+            )
+    else:
+        raise NotImplementedError(
+            "Support has not been implemented for api_url."
+        )
+
+    client_kwargs = {}
+    if "data" in llm_api_params:
+        assert isinstance(llm_api_params["data"], dict)
```

Comment:
`assert` should not be used in production code as it can be ignored depending on python interpreter flags. See [link](https://stackoverflow.com/questions/944592/best-practice-for-using-assert).

I would suggest changing this to `if not is_instance(..., dict): raise ...`